packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1474,17 +1526,25 @@ class _RenderDecoration extends RenderBox {
 
     if (label != null) {
       final double labelX = _boxParentData(label!).offset.dx;
+      final bool centerFloatLabel = decoration.floatingLabelAlignment == FloatingLabelAlignment.center;
```

Comment:
Instead of checking for `FloatingLabelAlignment.center`, could we calculate this based on `FloatingLabelAlignment.x`?  Here and below.

I know it's currently not possible to set it to anything else, but in the future if someone changes FloatingLabelAlignment to allow arbitrary values or to have another constant, they will expect it to be laid out correctly.

Reply:
I have strongly linked `FloatingLabelAlignment.x` with position of the label, so if right is needed, a new PR can be raised, adding a const to `FloatingLabelAlignment` and tests.

A problem I encountered: `InputDecoration.floatingLabelAlignment` defaults to `FloatingLabelAlignment.left`, which is semantically different from what is expected when text direction is `rtl`. The user may expect the label to be on the left end, but it will be on the right end instead. A solution to this would be to either change the semantics for consts in `FloatingLabelAlignment`, or set the default value according to text direction. lemme know your thoughts.

Reply:
Good point about RTL.  Instead of `left` and `right` (if/when we add it), we could use `start`, meaning left in LTR and right in RTL, and `end` meaning the opposite.

But do RTL users actually expect the label to be on the right?  Are there any guidelines for that in the Material spec or anything?

Reply:
The [bidirectionality](https://material.io/design/usability/bidirectionality.html) guideline states that the UI should be mirrored. Also, referencing the first do/don't [image](https://lh3.googleusercontent.com/GB88zPfVTyVngNcLRmhAtE6bAb9-4pCpCC__YymcXGNMRwvbzUQcG52SvKC6BivrF-4zxV3z-u9IXhLFkR37pKvvX86eD1AawbiOztM=w1064-v0) also has the floating label on the right.

Reply:
Thanks for looking that up.  It looks like Flutter actually does follow that guideline already, so fields with RTL directionality do put their label on the right.

How does FloatingLabelAlignment interact with this?  Like if `FloatingLabelAlignment.x` is 0.0 and the field is RTL, does the label still go all the way on the right?  You should add that as a test case.  Does it make sense that `x` reverses itself when in RTL mode?

At least it seems like we should rename `FloatingLabelAlignment.left` to `FloatingLabelAlignment.start` if indeed `0.0` can be right or left.

Reply:
(assuming by 0.0 you mean -1.0)

The FloatingLabelAlignment aligns the floating label independent of text direction. if set to left (x=-1.0), it'll align the floating label to the left end irrespective of text direction, and if set to center (x=0.0), it'll align to the center.

> Does it make sense that x reverses itself when in RTL mode?

That can be done if we use (start, center) instead of (left, center).

Okay, I'll update `left` to `start`, reverse x (just multiply it with -1) if text direction is rtl, and add relevant documentation.